react/features/calendar-sync/middleware.js

```diff
@@ -262,27 +267,58 @@ function _parseCalendarEntry(event, knownDomains) {
  * @returns {void}
  */
 function _updateCalendarEntries(events) {
-    if (events && events.length) {
-        // eslint-disable-next-line no-invalid-this
-        const { dispatch, getState } = this;
-
-        const knownDomains = getState()['features/base/known-domains'];
-        const eventList = [];
-
-        const now = Date.now();
-
-        for (const event of events) {
-            const calendarEntry = _parseCalendarEntry(event, knownDomains);
+    if (!events || !events.length) {
+        return;
+    }
 
-            calendarEntry
-                && calendarEntry.endDate > now
-                && eventList.push(calendarEntry);
+    // eslint-disable-next-line no-invalid-this
+    const { dispatch, getState } = this;
+    const knownDomains = getState()['features/base/known-domains'];
+    const now = Date.now();
+    const entryMap = new Map();
+
+    for (const event of events) {
+        const entry = _parseCalendarEntry(event, knownDomains);
+
+        if (entry && entry.endDate > now) {
+            // As was stated above, we don't display subsequent occurrences of
+            // recurring events, and the repetitions of events coming from
+            // multiple calendars.
+            const key = md5.hex(JSON.stringify({
```

Comment:
In all fairness, if the object to be used for the MD5 is to be as small as possible (as its properties suggest), why didn't you go with an array then?